doc/specs/File-diff-spec.md

```diff
@@ -0,0 +1,41 @@
+# New PowerToys utility: File comparison tool
+
+<img align="right" height="200px" src="https://github.com/microsoft/PowerToys/blob/user/aaron-junker/diff-spec/doc/images/icons/PowerToys%20icon/PNG/1440x2160.png" />
+
+- **What is it:** File comparison tool
+- **Authors:** [Aaron Junker](https://github.com/aaron-junker)
+- **Spec Status:** Draft
+- **GitHub issue:** [#14950](https://github.com/microsoft/PowerToys/issues/14950)
+
+## 1 Overview
+
+This utility will add a tool to view differences between two files.
+
+### 1.1 Technical implementation
+
+This utility will use [Microsoft Monaco Render engine](https://microsoft.github.io/monaco-editor/) which is currently implemented in the source code previewer. We will have to move the Monaco Editor source code and the corresponding helper classes to a new centralised 
+
+### 1.2 Use cases
+
+## 2 Goals and non-Goals
+
+### 2.1 Goals

```

Comment:
- local file comparison
- Would love if we could detect branches for git (could be URL as solution as a short term) vs local
- Binary files via hex editor?
- More than 2 files
- Ability to do Side by Side but also inline diff (GitHub vs GitHub Desktop)

Reply:
> Binary files via hex editor?
I guess for this we would have to create our own one.

> More than 2 files

Monaco is not capable of doing this. We can add it to the spec, but it shouldn't be a high priority item.

Reply:
* Three-way-diff support would be nice.
* Ability to label the files from the commandline (often my files will all have the same name, but, I might want the title to say `index.md (January)` and `index.md (March)`)